03_Modules/Transactions/src/module/module.ts

```diff
@@ -513,7 +513,9 @@ export class TransactionsModule extends AbstractModule {
         { totalKwh: totalKwh },
         { where: { id: transactionDbId }, returning: false },
       );
-      totalCost = this._roundCost(totalKwh * tariff.price);
+      // TODO: fix
+      let price = tariff.elements[0].priceComponents.find(component => component.isEnergy())!.price;
```

Comment:
Is it okay to fetch the price from the energy component of the first element? I'm uncertain about how to proceed at this point.